serverless-cljs-plugin/serverless_lumo/build.cljs

```diff
@@ -10,21 +10,37 @@
 (defn zip!
   "Create the zip in output-dir. Return a promise containing the path.
 
-  The dirs will be included in the zip, the compression level defaults
-  to 9."
+  The zip-opts map is:
+
+     {:dirs #{\"dir1\" [\"dir2\" false]} ;; https://archiverjs.com/docs/Archiver.html#directory
+      :files #{\"path/to/file\"}}        ;; https://archiverjs.com/docs/Archiver.html#file
+
+  Seqable entries are passed as subsequent parameters to the archiver
+  functions."
   [output-path zip-opts compiler-opts]
 
   (let [archiver      (archiver "zip" #js {:zlib {:level 9}})
         output-stream (.createWriteStream fs output-path)]
     (js/Promise.
      (fn [resolve-fn reject-fn]
+       (js/console.log "zipping %s..." output-path)
        (.on output-stream "close" #(resolve-fn output-path))
+       (.on archiver "warning" #(if (= (.-code %1) "ENOENT")
+                                  (js/console.warn (.-message %1))
+                                  (reject-fn %1)))
        (.on archiver "error" reject-fn)
 
        (.pipe archiver output-stream)
-       (doseq [d (zip-opts :dirs)]
-         (.directory archiver d))
-       (.file archiver (zip-opts :index) #js {:name "index.js"})
+       (doseq [d (:dirs zip-opts)]
```

Comment:
Please remove the log messages, it makes the code noisy and is not particularly helpful here.